src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -170,27 +213,62 @@ object Scalajsld {
 
       val linker = StandardImpl.linker(config)
       val logger = new ScalaConsoleLogger(options.logLevel)
-
-      val output = {
-        val js = options.output.toPath()
-        val sm = js.resolveSibling(js.getFileName().toString() + ".map")
-
-        def relURI(f: Path) =
-          new URI(null, null, f.getFileName().toString, null)
-
-        LinkerOutput(PathOutputFile(js))
-          .withSourceMap(PathOutputFile(sm))
-          .withSourceMapURI(relURI(sm))
-          .withJSFileURI(relURI(js))
-      }
-
       val cache = StandardImpl.irFileCache().newCache
 
       val result = PathIRContainer
         .fromClasspath(classpath)
-        .flatMap(containers => cache.cached(containers._1))
-        .flatMap(linker.link(_, moduleInitializers, output, logger))
+        .map(_._1)
+        .flatMap(cache.cached _)
+        .flatMap{irFiles =>
+          outputFile match {
+            case OutputFile.SingleFile(jsFile) =>
+              DeprecatedFileOutputSilencer().link(linker, irFiles.toList, moduleInitializers, jsFile, logger)
+            case OutputFile.Directory(outputDir) =>
+              linker.link(irFiles, moduleInitializers, PathOutputDirectory(outputDir.toPath), logger)
+          }
+        }
       Await.result(result, Duration.Inf)
     }
   }
+
+  private def createOutputFile(options: Options): Option[OutputFile] = {
+    options.output.map(OutputFile.SingleFile) orElse
+      options.outputDir.map(OutputFile.Directory)
+  }
+
+  private abstract class DeprecatedFileOutputSilencer {
+    def link(linker: Linker,
+             irFiles: Seq[IRFile],
+             moduleInitializers: Seq[ModuleInitializer],
+             linkerOutputFile: File, logger: Logger)(
+              implicit ec: ExecutionContext): Future[Unit]
+  }
+
+  private object DeprecatedFileOutputSilencer extends DeprecatedFileOutputSilencer {
+    def apply(): DeprecatedFileOutputSilencer = this
+
+    @deprecated("Deprecate to silence warnings", "never/always")
+    def link(linker: Linker,
+             irFiles: Seq[IRFile],
+             moduleInitializers: Seq[ModuleInitializer],
+             linkerOutputFile: File, logger: Logger)(
+              implicit ec: ExecutionContext): Future[Unit] = {
```

Comment:
Indentation here is not Scala.js style. Have a look here: https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#indentation

Reply:
I missed this doc. I missed this ugly line break, too.